### PR TITLE
Make container images configurable in the CR

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
Resolves: #100

Currently we have a unit test for the default values. Do we need another unit test for the entire defaulter webhook?

### CODE REVIEW

File: apis/tempo/v1alpha1/microservices_types.go [hunk 1]
```diff
@@ -94,6 +100,18 @@ type MicroservicesList struct {
 	Items           []Microservices `json:"items"`
 }
 
+type ImagesSpec struct {
```
Comment (frzifus) [inline apis/tempo/v1alpha1/microservices_types.go:103]: lets put a comment on top and describe what its used for.

File: main.go [hunk 2]
```diff
@@ -79,6 +79,7 @@ func main() {
 			setupLog.Error(err, "unable to create webhook", "webhook", "Microservices")
```
Comment (frzifus) [inline main.go:79]:
```suggestion
			setupLog.Error(err, "unable to create validator webhook", "webhook", "Microservices")
```

File: main.go [hunk 3]
```diff
@@ -79,6 +79,7 @@ func main() {
...
 			os.Exit(1)
 		}
+		mgr.GetWebhookServer().Register("/mutate-tempo-grafana-com-v1alpha1-microservices", tempov1alpha1.NewMicroservicesDefaulterWebhook(mgr.GetClient(), ctrlConfig.DefaultImages))
```
Comment (frzifus) [inline main.go:82]: Does register return an error that we should handle?

**Review by frzifus — COMMENTED** [review]
i like :)

**Comment by andreasgerstmayr** [comment]
Just after opening the PR I realized afaics I can also use a custom defaulter:
```
return ctrl.NewWebhookManagedBy(mgr).
		For(r).
		WithDefaulter(&defaulter{defaultImages: defaultImages}).
		Complete()
```
saving us a bit of boilerplate code in handling the webhook. I'll update this PR shortly.

**Comment by andreasgerstmayr** [comment]
Updated the PR with a custom defaulter, so we can keep using the webhook builder, get a generated webhook path and in general need less boilerplate.

**Comment by andreasgerstmayr** [comment]
Fixed the linter error.

Thanks for the review  @frzifus!

File: apis/tempo/v1alpha1/microservices_webhook.go [hunk 4]
```diff
@@ -11,28 +15,50 @@ import (
 )
 
 var (
-	zeroQuantity  = resource.MustParse("0Gi")
-	tenGBQuantity = resource.MustParse("10Gi")
+	zeroQuantity                = resource.MustParse("0Gi")
+	tenGBQuantity               = resource.MustParse("10Gi")
+	ErrNoDefaultTempoImage      = errors.New("please specify a tempo image in the CR or in the operator configuration")
+	ErrNoDefaultTempoQueryImage = errors.New("please specify a tempo-query image in the CR or in the operator configuration")
```
Comment (frzifus) [inline apis/tempo/v1alpha1/microservices_webhook.go:21]:
```suggestion
	errNoDefaultTempoImage      = errors.New("please specify a tempo image in the CR or in the operator configuration")
	errNoDefaultTempoQueryImage = errors.New("please specify a tempo-query image in the CR or in the operator configuration")
```

We do not use this errors somewhere else, right? If so, i would recommend to keep them private.

**Review by frzifus — APPROVED** [review]
awesome :) Ive only one tiny thing ^^

File: apis/tempo/v1alpha1/microservices_webhook.go (hunk 4 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/microservices_webhook.go]: Good idea, I've updated the PR.

**Pull request merged**